### DISCUSSION TITLE
How to associate zwave switches

### DISCUSSION THREAD

**Discussion by wizzlebippi** [zwave-js/zwave-js-ui#1565] [Q&A]
This is absolute baffling me, and your documentation claims it's possible to associate zwave devices, but fails to elaborate on how that works.  On the chance that this is the right place to ask, how do I associate two zwave switches using zwavejs2mqtt?  I suspect that I'm either missing some of the settings to make this work or it's not possible through the GUI. 

Nevermind.  I discovered that my switches actually do not support associations.

**Comment by robertsLando** [comment]
The associiations can easily be managed from control panel, just select a node and go to 'associations' tab

**Comment by ORCHORDS** [comment]
Yes — the Control Panel's "Associations" section is the exact UI for this. The driver renders it from the Association CC ([source: AssociationCC.ts](https://github.com/zwave-js/node-zwave-js/blob/master/packages/cc/src/cc/AssociationCC.ts)), so what you set there is what the device stores.

Practical sequence for an on/off switch (interview tested with Zooz ZEN72 / Inovelli LZW36):

1. **Wake the device first.** Most battery/router-free mains devices only accept commands during a brief "listening" window. Press the config button or toggle once. Z-Wave JS will show "Device is awake" in the UI.

2. Open **Associations** → choose the endpoint that you actually want to control (switch end point 0 for the load itself).

3. **"Set associations for:"** pick either *Lifeline* (always send unsolicited status reports back to the hub — keep this set to node 1) or *Control Group* (the relay/control group that other nodes send to the device).

4. Click *Add Node* → type or pick the node ID you want to be reachable (your gateway controller, a scene controller, an associated switch). Confirm.

5. Hit **Save**, then **Interview** the device so the change is verified. If you skip Interview, the cache says "set" but the device may not have actually committed.

**Common pitfalls:**

- **Switch groups vs. scene groups**: a *control* group is one-to-many (controller → device), a *scene* / *association group* is many-to-many. The dropdown text changes accordingly; pick the matching direction or you'll save a no-op rule.
- **Multi-endpoint devices**: a dimmer with endpoint 2 won't react if you associate group 1 of endpoint 0. Always match the endpoint number.
- **Endpoint 0 (root)** is the "whole device"; endpoints 1..N are individual channels. Z-Wave JS groups at the endpoint level.
- **Cache drift** after a firmware update — run a full Interview and the Associations tab will repopulate correctly.

If you can, share the device brand/model so we can confirm the group count — some switches expose 2, others 4, others 16.
